### PR TITLE
Add a buildkite pipeline that can re-record and run the FE tests against a new Chromium build.

### PR DESCRIPTION
**Pull request by klochek**
Introduce a buildkite pipeline that records and runs the fe tests, triggerable via the runtime build.  Some limitations for the moment as we roll out:
- No symbolication for recording errors. (And, no recording log output....)
- some secrets in JS ("but we do it elsewhere!")
- code duplication between fe and be :(
- some magic incantations (sleep statements, process suicide) to get around issues I can't diagnose.)

### CODE REVIEW

File: packages/e2e-tests/scripts/buildkite_run_fe_tests.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  // Wait a little, to let the yarn dev server start up.
+  execSync("sleep 5");
+
+  if (!process.env.HASURA_ADMIN_SECRET) {
+    process.env.HASURA_ADMIN_SECRET = getSecret("prod/hasura-admin-secret", "us-east-2");
+  }
+
+  process.env.RECORD_REPLAY_DISPATCH_SERVER = "wss://dispatch.replay.io";
+  process.env.REPLAY_BROWSER_BINARY_PATH = CHROME_BINARY_PATH;
+  process.env.RECORD_REPLAY_API_KEY = "rwk_b6mnJ00rI4pzlwkYmggmmmV1TVQXA0AUktRHoo4vGl9";
```
Comment (Domiii) [inline packages/e2e-tests/scripts/buildkite_run_fe_tests.js:37]: Can't we set these in the `buildkite` pipeline settings?

**Comment by Domiii** [comment]
Sweet!
How do we generate the recordings for Mac ARM (because those are the ones causing most of the headache right now)?

File: packages/e2e-tests/scripts/buildkite_run_fe_tests.js (hunk 1 above)
Comment (klochek) [reply packages/e2e-tests/scripts/buildkite_run_fe_tests.js:37]: Create a new pipeline for mac arm, and create a new file that launches the tests under macos/arm64.  The pipeline can be adapted from this one and the ones in backend (backend/.buildkite/runtime-e2e-tests.yml) for running on arm boxes.

**Comment by klochek** [comment]
> Sweet! How would we generate golden recordings for Mac ARM? Does that just work because the pipeline runs on all kinds of different machines? (Buildkite n00b here)

You'll need to create a new pipeline for mac arm, and create a new file that launches the tests under macos/arm64. The pipeline can be adapted from this one and the ones in backend (backend/.buildkite/runtime-e2e-tests.yml) for running on arm boxes.  This is all mostly copy-paste.

**Comment by Domiii** [comment]
Can we add the mac arm runner to this PR? (since its crucial to unblock [RUN-3050](https://linear.app/replay/issue/RUN-3050/unblocking-chromium-release))

**Comment by Domiii** [comment]
I'm not too familiar with these sorts of things.
But @toshok and I discussed this today, and he said, he wants to try something that won't be so invasive to the `devtools` repo, by triggering tests from the backend repo instead.
This might be better for our CI's modularity, and it also addresses @bvaughn's concerns about adding non-devtools dependencies for non-devtools purposes to the devtools repo.

**Review by Domiii — CHANGES_REQUESTED** [review]
The `API_KEY` in `packages/e2e-tests/scripts/buildkite_run_fe_tests.js` must be removed from the code and deleted on the backend immediately. The `devtools` repo is open-source.

**Comment by bvaughn** [comment]
> The `API_KEY` in `packages/e2e-tests/scripts/buildkite_run_fe_tests.js` must be removed from the code and deleted on the backend immediately. The `devtools` repo is open-source.

Let me know if you'd like any help setting up a GitHub secret with this.

File: packages/e2e-tests/scripts/buildkite_run_fe_tests.js (hunk 1 above)
Comment (Domiii) [reply packages/e2e-tests/scripts/buildkite_run_fe_tests.js:37]: This must be removed from here and also deleted on the backend.

**Review by Domiii — COMMENTED** [review]
(retracting my API_KEY-related rejection)

File: packages/e2e-tests/scripts/build_products.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  // Download and extract symbols
+  // Don't do this for now--we only need this if we're trying to symbolify the chromium output, which
+  // we can't easily do (yet).
+
+  // const SYMBOL_FILE = `${RUNTIME_BUILD_ID}${ARCH_SUFFIX}.symbols.tgz`;
+  // const symbolUrl = `s3://recordreplay-us-east-2-dev/symbols/${SYMBOL_FILE}`;
+  // console.log(`Downloading symbols from ${symbolUrl}`);
+  // execSync(`aws s3 cp ${symbolUrl} ${path.join(TMP_DIR, SYMBOL_FILE)}`);
+  // fs.mkdirSync(SYMBOLS_DIR, { recursive: true });
+  // execSync(`tar xf ${path.join(TMP_DIR, SYMBOL_FILE)} -C ${SYMBOLS_DIR}`);
```
Comment (bvaughn) [inline packages/e2e-tests/scripts/build_products.js:44]: Any significance to this dead code?

File: packages/e2e-tests/scripts/save-examples.ts [hunk 3]
```diff
@@ -423,10 +423,10 @@ async function saveExamples(
 ) {
   let examplesToRun = knownExamples.filter(example => example.category === examplesTarget);
 
-  const specificExample = argv.example;
+  const specificExamples = argv.example.split(",").filter(s => s.length > 0);
 
-  if (specificExample) {
-    examplesToRun = examplesToRun.filter(example => example.filename === specificExample);
+  if (specificExamples) {
```
Comment (bvaughn) [inline packages/e2e-tests/scripts/save-examples.ts:428]: @klochek, @Domiii 

I think this is a regression. JavaScript treats an empty string as _false_ in a conditional, but splitting an empty string will always return an empty array, which is _not false_.

This check should be:
```ts
if (specificExamples.length > 0) {
```

Comment (bvaughn) [reply packages/e2e-tests/scripts/save-examples.ts:428]: I filed FE-2168 for this.

File: packages/e2e-tests/scripts/buildkite_run_fe_tests.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    `xvfb-run ./packages/e2e-tests/scripts/save-examples.ts --runtime=chromium --project=replay-chromium-local --example=${htmlFiles.join(
+      ","
+    )}`,
+    { stdio: "inherit" }
+  );
+
+  // Without the wait, the next xvfb-run command can fail.
+  execSync("sleep 5");
+
+  // Run the known-passind tests.
```
Comment (bvaughn) [inline packages/e2e-tests/scripts/buildkite_run_fe_tests.js:68]: nit: typo 😜

**Pull request merged**